src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,48 +1,92 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo,
+  clickModal: () => void
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, clickModal }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = todo;
+
+  useEffect(() => {
+    const loadUser = async () => {
+      try {
+        const loadedUser = await getUser(userId);
```

Comment:
```suggestion
        setIsLoading(true);

        const loadedUser = await getUser(userId);
```
you should set isLoading `true` right before starting loading user